### PR TITLE
test(storage): add WriteTo integration test cases

### PR DESCRIPTION
**Pull request by tritone**
Update the most important integration tests for Reader to cover both Read and WriteTo.

Also remove refs to deprecated ioutil.

### CODE REVIEW

File: storage/integration_test.go [hunk 1]
```diff
@@ -2578,10 +2606,22 @@ func TestIntegration_ZeroSizedObject(t *testing.T) {
...
+				}
+				body, err := c.readFunc(r)
+				if err != nil {
+					t.Fatalf("reading object: %v", err)
+				}
+				if len(body) != 0 {
+					t.Errorf("Body is %v, want empty []byte{}", body)
+				}
+			})
+
```
Comment (BrennaEpp) [inline storage/integration_test.go:2624]: nit: this empty line can probably be removed

File: storage/integration_test.go [hunk 2]
```diff
@@ -291,6 +290,29 @@ func multiTransportTest(ctx context.Context, t *testing.T,
...
+	desc     string
+	readFunc (func(io.Reader) ([]byte, error))
+}
+
+var readCases = []readCase{
+	{
+		desc: "Read",
+		readFunc: func(r io.Reader) ([]byte, error) {
+			return io.ReadAll(r)
+		},
```
Comment (BrennaEpp) [inline storage/integration_test.go:304]:
```suggestion
		readFunc: io.ReadAll,
```

File: storage/integration_test.go [hunk 3]
```diff
@@ -3587,23 +3627,33 @@ func TestIntegration_ReadCRC(t *testing.T) {
...
-			}
-			_, err = ioutil.ReadAll(r)
-			_ = r.Close()
-			if err != nil {
-				t.Fatalf("%s: %v", test.desc, err)
-			}
+
+			})
 		}
+
```
Comment (BrennaEpp) [inline storage/integration_test.go:3656]: nit:
```suggestion
```

File: storage/integration_test.go [hunk 4]
```diff
@@ -4472,49 +4522,57 @@ func TestIntegration_Reader(t *testing.T) {
...
+						_, err = rc.Read(buf)
+						if err != nil {
+							t.Fatalf("%v: %v", obj, err)
+						}
+						if got, want := buf, contents[obj][:1]; !bytes.Equal(got, want) {
+							t.Errorf("Contents[0] (%q) = %q; want %q", obj, got, want)
+						}
+						if err := rc.Close(); err != nil {
+							t.Errorf("%v Close: %v", obj, err)
+						}
```
Comment (BrennaEpp) [inline storage/integration_test.go:4571]: Suggestion: Can this be moved out one level? I don't think we need to test this for both read funcs.

**Review by BrennaEpp — APPROVED** [review]
A couple suggestions but otherwise LGTM

File: storage/integration_test.go (hunk 4 above)
Comment (tritone) [reply storage/integration_test.go:4571]: I think it's fine to leave it as is; it doesn't shrink the test matrix at all to make this change.

**Pull request merged**